### PR TITLE
Send kafka message in batch

### PR DESCRIPTION
**Pull request by RoySunnySean007**
According to the discussion in https://community.k6.io/t/suggestion-send-message-in-batch-for-k6-kafka-output/1068

### CODE REVIEW

**Comment by MStoykov** [comment]
Hi @RoySunnySean007 , can you:
1. delete instead of commenting the code
2. fix the linter errors 
3. give some numbers on how much of an improvement this makes. I am not sayign it doesn't have an improvement, but it will be nice for it to be measured so we can put them in the release notes.

Also unfortunately we release v0.29.0 on Monday ... by plan .. hopefully. And this just comes way too late to be included, probably. So I am just going to mark this for v0.30.0 and hopefully we will merge it early on in the release cycle :)

**Comment by mostafa** [comment]
Hey @RoySunnySean007,

Nice job! :+1: How will this bulk sending of messages to Kafka deal with logging of any sorts, since you've effectively removed the logging code?

**Comment by RoySunnySean007** [comment]
Dear @MStoykov  & @mostafa ,

1. I have updated the patch file.
2. This change was made one year ago on v0.26 by my colleague. Based on our memory, it will take 30 mins to finish sending all the k6 raw data result to kafka, load test tps should be less than 1k/s. After we use batch mode, there is no delay.

Sorry we raised this PR a bit late...we lost some data...

Thanks
Roy

**Comment by RoySunnySean007** [comment]
Dear @mostafa, 

Could u pls kindly help to review it? Any comment is appreciated, thx!

File: stats/kafka/collector.go [hunk 1]
```diff
@@ -155,19 +155,15 @@ func (c *Collector) pushMetrics() {
...
 	// Send the samples
 	c.logger.Debug("Kafka: Delivering...")
-
+	
+	var msgList []*sarama.ProducerMessage
 	for _, sample := range formattedSamples {
 		msg := &sarama.ProducerMessage{Topic: c.Config.Topic.String, Value: sarama.StringEncoder(sample)}
-		partition, offset, err := c.Producer.SendMessage(msg)
-		if err != nil {
-			c.logger.WithError(err).Error("Kafka: failed to send message.")
```
Comment (mostafa) [inline stats/kafka/collector.go:163]: Is there any way to have logging while using `c.Producer.SendMessages(msgList)` on Line 165?

File: stats/kafka/collector.go [hunk 2]
```diff
@@ -155,19 +155,15 @@ func (c *Collector) pushMetrics() {
...
-		} else {
-			c.logger.WithFields(logrus.Fields{
-				"partition": partition,
-				"offset":    offset,
-			}).Debug("Kafka: message sent.")
-		}
+		msgList = append(msgList, msg)
 	}
+	// Send message in batch
+	c.Producer.SendMessages(msgList)
```
Comment (mostafa) [inline stats/kafka/collector.go:165]: > Is there any way to have logging while using `c.Producer.SendMessages(msgList)` on Line 165?

That is, what happens if this line somehow fails?

Comment (RoySunnySean007) [reply stats/kafka/collector.go:165]: Good point! Let me have a check. Thx @mostafa !

Comment (MStoykov) [reply stats/kafka/collector.go:165]: Hi @RoySunnySean007 any movement on this change?

**Comment by MStoykov** [comment]
Thanks for all the work done here, but given that we are:
1. [depricateing kafka output](https://github.com/k6io/k6/issues/1934)
2. have moved it out as an [extension](https://github.com/k6io/xk6-output-kafka)

I will close this PR and ask that if you want to add this functionality to do so in the extension repo.

**Pull request closed without merging**